app/src/main/java/com/hamdan/forzenbook/data/repository/LoginRepositoryImpl.kt

```diff
@@ -4,15 +4,18 @@ import com.hamdan.forzenbook.data.database.LoginDao
 import com.hamdan.forzenbook.data.network.LoginService
 
 class LoginRepositoryImpl(
+    // TODO implement the dao FA-84
     private val dao: LoginDao,
     private val service: LoginService
 ) : LoginRepository {
     override suspend fun requestValidation(email: String): Boolean {
         return service.requestValidation(email).isSuccessful
     }
 
-    // TODO Implement Getting the Token
     override suspend fun getToken(email: String, code: String): LoginData? {
-        TODO("Not yet implemented")
+        val response = service.getToken(email, code)
+        return if (response.isSuccessful) {
+            LoginData(response.body()?.token)
```

Comment:
Shouldn't this just be returning the body? The token is just a string in the JSON

Reply:
it definitely could be, I just was following from our earlier lessons in that we wrapped our info from the body in a data object, it did feel a little pointless for this since it was a singular item.